finesse/gui/opus_view.py

```diff
@@ -0,0 +1,226 @@
+        return layout
+
+    def url(self, action: str) -> str:
+        """Builds an URL out of the ip and the action.
+
+        Args:
+            action: The action to use.
+
+        Returns:
+            str: The constructed URL
+        """
+        return f"http://{self.ip}/{self.commands[action]}"
+
+    def on_acction_button_clicked(self, action: str) -> None:
+        """Execute the given action by sending a message to the appropriate topic.
+
+        TODO: Here assuming we are going to use pubsub or equivalent to send messages
+        around and communicate between backend and front end.
+
+        Args:
+            action: Action to be executed.
+        """
+        logging.info(f"OPUS action '{self.url(action)}' executed!")
+
+    def display_status(self) -> None:
+        """Retrieves and displays the new status.
+
+        TODO: I'm not sure who should populate the error log in the GUI. Probably the
+        ones handling the individual actions above. These are all placeholders, for now.
+        """
+        logging.info("Getting OPUS status!")
+        self.status.load(QUrl(self.url("status")))
+        self.status.show()
+
+        logging.getLogger("OPUS").error("Oh, no! Something bad happened!")
+
+    def open_opus(self) -> None:
+        """Opens OPUS front end somewhere else.
+
+        TODO: No idea what this is supposed to do.
+        """
+        logging.info("Going to OPUS!")
+
+
+class OPUSLogHandler(logging.Handler):
+    """Specific logger for the errors related to OPUS.
+
+    Only log messages using the OPUS logger will be recorded here. Typically, they will
+    be error messages, but it can be any information worth to be logged.
+    """
+
+    @classmethod
+    def set_handler(cls, log_area: QTextBrowser):
+        """Creates the handler and adds it to the logger.
+
+        Args:
+            log_area: The area where the log will be printed.
+        """
+        ch = cls(weakref.ref(log_area))
+
+        formatter = logging.Formatter(
+            fmt="%(asctime)s [%(levelname)s]: %(message)s", datefmt="%Y/%m/%d %H:%M:%S"
+        )
+        ch.setFormatter(formatter)
+
+        logging.getLogger("OPUS").addHandler(ch)
+
+    def __init__(self, log_area: weakref.ref):
+        """Constructor of the Handler.
+
+        Args:
+            log_area: A weak reference to the log area.
+        """
+        super(OPUSLogHandler, self).__init__()
+        self.log_area = log_area
+
+    def emit(self, record):
+        """Add the record to the text area.
+
+        If the log area has been destroyed before the logger, it will raise an
+        AttributeError. This can only happen during tests and can be safely ignored.
+        """
+        try:
+            self.log_area().append(self.format(record))
+            cursor = self.log_area().textCursor()
+            cursor.movePosition(QTextCursor.End)
+            self.log_area().setTextCursor(cursor)
+        except AttributeError:
+            pass
```

Comment:
I'd just convert `log_area` to a strong reference once at the start of the function, rather than doing it every time you want it:

```suggestion
        log_area = self.log_area()
        if not log_area:
            # log_area no longer exists
            return

        log_area.append(self.format(record))
        cursor = log_area.textCursor()
        cursor.movePosition(QTextCursor.End)
        log_area.setTextCursor(cursor)
```

Reply:
I'm not sure if this will work as you think because what happens is - I think - the python wrapper to the underlying C++ object still exists (so self.log_area is not None) but the actual C++ object is done. If it works, you option is way cleaner. Let's use it and see how it goes.